### PR TITLE
Restructure examples directory

### PR DESCRIPTION
**Pull request by ChumpChief**
The examples directory has become less organized over time as examples have been added, removed, or modified.  This change does a few things:

1. Eliminates the "hosts" directory.  This never really had a distinct meaning from "apps" anyway.
2. Declares "data-objects" to be specifically for webpack-fluid-loader examples (examples not using webpack-fluid-loader either move to "apps" or a more specific location).  Added a README to indicate this.
3. Sets up several subdirectories for more-specific and meaningful topic groupings:
    * benchmarks
    * external-data
    * version-migration
    * view-integration
4. Moves examples under /experimental into the /examples directory in their appropriate location. /experimental/examples is eliminated.
5. Rescopes `@fluid-experimental/inventory-app` to `@fluid-example/inventory-app`
6. Moves `@fluid-experimental/tree-react-api` to /experimental/framework (similar to react-inputs)
7. Fixes up some incorrect `repository.directory` entries in package.jsons

Not done in this PR:
1. multiview should really stop using webpack-fluid-loader and be moved to the view-integration directory, but I think it's better to remain where it is until that work is done.
2. In general, my opinion is that we should stop using webpack-fluid-loader in all cases.  I don't do any conversion in this change though.

### CODE REVIEW

File: azure/packages/external-controller/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	"repository": {
 		"type": "git",
 		"url": "https://github.com/microsoft/FluidFramework.git",
-		"directory": "examples/hosts/app-integration/external-controller"
+		"directory": "azure/packages/external-controller"
```
Comment (Abe27342) [inline azure/packages/external-controller/package.json:10]: could you double-check scripts in each changed package? It looks like the prettier commands use relative paths dependent on package location and some of these moves change the number of folders.

side-note: I'm not sure why we do that or if it's necessary. I'd expect that there should be a way to set prettier up to be more resilient to package moves like this.

**Review by Abe27342 — COMMENTED** [review]
looks good modulo the scripts comment. feel free to ping me once you've looked into that and I can approve.

File: azure/packages/external-controller/package.json (hunk 1 above)
Comment (ChumpChief) [reply azure/packages/external-controller/package.json:10]: Good catch - maybe .prettierignore could move under `@fluidframework/build-common`, but for this change I'll just fix up the depths.

Comment (ChumpChief) [reply azure/packages/external-controller/package.json:10]: I see this also bit recently with #15355

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**